sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/MergingActiveWindowSet.java

```diff
@@ -124,7 +126,20 @@ public MergingActiveWindowSet(WindowFn<Object, W> windowFn, StateInternals<?> st
   }
 
   @Override
-  public void removeEphemeralWindows() {
+  public void garbageCollect() {
+    // Any remaining NEW windows must have been for closed windows and can be forgotten (again).
+    List<W> stillNew = new ArrayList<>();
```

Comment:
I don't think you need to make the copy. If you're trying to avoid concurrent modification, I think you can use the Iterator explicitly and call Iterator#remove, which removes it from the map (when used on the entrySet).

So then:
```
Iterator<Map.Entry<W, Set<W>>> iterator = activeWindowToStateAddressWindows.entrySet().iterator();
while (iterator.hasNext()) {
  Map.Entry<W, Set<W>> entry = iterator.next();
  if (entry.getValue().isEmpty()) {
    windowToActiveWindow.remove(entry.getKey());
    iterator.remove();
  }
}
```